src/classes/WalletConnectors/BrowserConnector.ts

```diff
@@ -0,0 +1,26 @@
+import { type Eip1193Provider } from 'ethers'
+import { WalletConnector } from './WalletConnector'
+
+export interface WindowProvider extends Eip1193Provider {
+    isMetaMask?: boolean
+    isTrust?: boolean
+    isSafePal?: boolean
+    isBitKeep?: boolean
+    providers?: WindowProvider[]
+}
+
+export class BrowserConnector extends WalletConnector<WindowProvider | undefined> {
+    public readonly id: string = 'browserConnector'
+    public readonly name: string = 'Browser Wallet'
+
+    public getProvider() {
+        if (typeof window === 'undefined') return
+
+        const ethereum = (window as unknown as { ethereum?: WindowProvider })
+            .ethereum
+
+        if (ethereum?.providers) return ethereum.providers[0]
```

Comment:
why return `provider` instead of ethereum?

Reply:
Some browser wallet extensions can construct a "multiprovider" array at `window.ethereum.providers` containing the injected provider from each wallet, e.g. CoinBase to avoid conflicts if there is already an injected provider on `window.ethereum`. In this case we check and return the first one of the list.

I've decided to create this code to have a template / snippet where we can use to extend other browser connectors, but if you think that this isn't the right place for this code let me know and we can think a better approach